datasets/metashift/metashift.py

```diff
@@ -0,0 +1,395 @@
+@InProceedings{liang2022metashift,
+title={MetaShift: A Dataset of Datasets for Evaluating Contextual Distribution Shifts and Training Conflicts},
+author={Weixin Liang and James Zou},
+booktitle={International Conference on Learning Representations},
+year={2022},
+url={https://openreview.net/forum?id=MTex8qKavoS}
+}
+"""
+
+
+_DESCRIPTION = """\
+The MetaShift is a dataset of datasets for evaluating distribution shifts and training conflicts.
+The MetaShift dataset is a collection of 12,868 sets of natural images across 410 classes.
+It was created for understanding the performance of a machine learning model across diverse data distributions.
+"""
+
+_HOMEPAGE = "https://metashift.readthedocs.io/"
+
+
+_LICENSE = "https://github.com/Weixin-Liang/MetaShift/blob/main/LICENSE"
+
+
+_URLS = {
+    "image_files": "https://nlp.stanford.edu/data/gqa/images.zip",
+    "scene_graph_annotations": "https://nlp.stanford.edu/data/gqa/sceneGraphs.zip",
+}
+
+_METADATA_URLS = {
+    "full_candidate_subsets": "https://github.com/Weixin-Liang/MetaShift/raw/main/dataset/meta_data/full-candidate-subsets.pkl",
+}
+
+_ATTRIBUTES_URLS = {
+    "attributes_candidate_subsets": "https://github.com/Weixin-Liang/MetaShift/raw/main/dataset/attributes_MetaShift/attributes-candidate-subsets.pkl",
+}
+
+
+# See https://github.com/Weixin-Liang/MetaShift/blob/main/dataset/meta_data/class_hierarchy.json
+# for the full object vocabulary and its hierarchy.
+# Since the total number of all subsets is very large, all of the following scripts only generate a subset of MetaShift.
+
+_CLASSES = [
+    "cat",
+    "dog",
+    "bus",
+    "truck",
+    "elephant",
+    "horse",
+]
+
+
+_ATTRIBUTES = [
+    "cat(orange)",
+    "cat(white)",
+    "dog(sitting)",
+    "dog(jumping)",
+]
+
+
+class MetashiftConfig(datasets.BuilderConfig):
+    """BuilderConfig for MetaShift."""
+
+    def __init__(
+        self,
+        selected_classes=None,
+        attributes_dataset=False,
+        attributes=None,
+        with_image_metadata=False,
+        image_subset_size_threshold=25,
+        min_local_groups=5,
+        **kwargs,
+    ):
+        """BuilderConfig for MetaShift.
+
+        Args:
+        selected_classes: `list[string]`, list of the classes needed to generate subsets.
+        attributes_dataset: `bool`, default=False, to generate the MetaShift-Attributes dataset, set as True
+        attributes: `list[string]`, list of attributes classes needed to generate the Attributes dataset.
+        with_image_metadata: `bool`, default=False, to include image metadata, set as True
+        image_subset_size_threshold: `int`, default=25, The number of images required to consider a subset.
+        If the number of images are less than image_subset_size_threshold, then the subset is not considered.
+        min_local_groups: `int`, default=5, The minimum number of local groups required to consider an object class.
+        **kwargs: keyword arguments forwarded to super.
+        """
```

Comment:
Please indent the params list here (after `Args:`) and update the docstring with the suggestions from the `Dataset Usage` section to keep these parts as aligned as possible.

Reply:
I've indented the params list and updated the docstring. I haven't included the links mentioned in the Dataset Usage section though, let me know if we need to add them as well.